### PR TITLE
YAML format change

### PR DESCRIPTION
**Pull request by kachulis**
Changed YAML format, as discussed in https://github.com/broadinstitute/palantir-workflows/pull/176#discussion_r1620768945

Also changed to use ruamel.yaml for yaml parsing so that duplicate keys will not be allowed (PyYAML overwrites duplicate keys with the last specified value)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-pyyaml==6.0.1
+ruamel.yaml==0.18.6
```
Comment (rickymagner) [inline requirements.txt:1]: Good call, thanks!

**Review by rickymagner — COMMENTED** [review]
Thanks for doing this refactor. Did you test with the decoy tests in this repo that the new format works with this layout?

**Comment by kachulis** [comment]
I tested the ones we expect to succeed, though not the ones we expect to fail.  My preference would merge #4, rebase, and then let the automatic testing take over.

File: .github/workflows/run_tests.yaml [hunk 2]
```diff
@@ -10,22 +10,22 @@ jobs:
           python watt.py -e cromwell-84.jar -t simple compress_file expect_fail -p $(nproc)
           echo "THE FOLLOWING TESTS SHOULD ALL FAIL"
           if python watt.py -e cromwell-84.jar -w say_hello -t mismatch; then
-            echo "A test succeeded which should have passed.  Please look above for an unexpected success"
+            echo "Teat mismatch for workflow say_hello succeeded which should not have passed."
```
Comment (rickymagner) [inline .github/workflows/run_tests.yaml]: Should be "Test mismatch ..."

**Pull request merged**